tests/aws/services/s3/test_s3.py

```diff
@@ -4251,16 +4257,21 @@ def test_create_bucket_head_bucket(self, snapshot, aws_client):
                 snapshot.transform.regex(r"s3\.amazonaws\.com", "<host>"),
                 snapshot.transform.regex(r"s3\.localhost\.localstack\.cloud:4566", "<host>"),
                 snapshot.transform.regex(r"s3\.localhost\.localstack\.cloud:443", "<host>"),
+                snapshot.transform.key_value("Location", reference_replacement=False),
             ]
         )
 
         try:
-            response = aws_client.s3.create_bucket(Bucket=bucket_1)
+            response = create_s3_bucket(bucket_name=bucket_1, s3_client=aws_client.s3)
```

Comment:
not sure we should use the `aws.resources.create_s3_bucket` here, maybe we could use a fixture instead of the big `try...finally` block. Not blocking though, this is okay for now but a consideration for the future.